src/commcare_cloud/environment/main.py

```diff
@@ -43,15 +50,35 @@ def check(self):
         self.proxy_config
         self.create_generated_yml()
 
-    @memoized
     def get_ansible_vault_password(self):
+        """Get ansible vault password
+
+        This method has a side-effect: it records a Datadog event with
+        the commcare-cloud command that is currently being run.
+        """
+        self.get_vault_variables()
+        return self._get_ansible_vault_password()
+
+    def get_vault_variables(self):
+        """Get ansible vault variables
+
+        This method has a side-effect: it records a Datadog event with
+        the commcare-cloud command that is currently being run.
```

Comment:
I took the side-effect approach because it seemed easier and cleaner than sprinkling `record_command_event` calls around in various places near where vault variables or the ansible password are loaded. An argument could be made that this is too sneaky and hacky. I'm open to feedback or other ideas.